Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -536,13 +536,15 @@ void HTMLInputElement::updateType(const AtomString& typeAttributeValue)
     m_inputType->detachFromElement();
     auto oldType = m_inputType->type();
 
+    bool didDirAutoUseValue = m_inputType->dirAutoUsesValue();
     bool previouslySelectable = m_inputType->supportsSelectionAPI();
 
     m_inputType = WTFMove(newType);
     m_inputType->createShadowSubtreeIfNeeded();
 
-    if (oldType == InputType::Type::Telephone || m_inputType->type() == InputType::Type::Telephone)
-        updateTextDirectionalityAfterTelephoneInputTypeChange();
+    if (oldType == InputType::Type::Telephone || m_inputType->type() == InputType::Type::Telephone
+        || (hasDirectionAuto() && didDirAutoUseValue != m_inputType->dirAutoUsesValue()))
```

Comment:
It might be nice to put this in a method that describes what this check is for. Up to you.

But at a minimum, we should probably include a link to the spec here?

Reply:
Added a spec link